<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Visualizing and Annotating Phylogenetic Trees with R+ggtree</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />
<!-- Font Awesome -->
<!-- <link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" /> -->
<!-- <script src="https://use.fontawesome.com/54ee8c2dfd.js"></script> -->

<!-- Google fonts -->
<link href='https://fonts.googleapis.com/css?family=Open+Sans:400,400italic,700,700italic|Oswald:400,700' rel='stylesheet' type='text/css'>

<!-- Favicon -->
<link rel="shortcut icon" type="image/x-icon" href="img/favicon.ico">

<!-- Google analytics -->
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-8298649-8', 'auto');
  ga('send', 'pageview');
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css" type="text/css" />
<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Bionesia2021</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fas fa-home"></span>
     
    Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa-cog"></span>
     
    Intro to R
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="setup.html">Plot in R</a>
    </li>
    <li>
      <a href="r-ggtree_edit.html">Ggtree</a>
    </li>
    <li>
      <a href="r-ggtree.html">Ggtree1</a>
    </li>
  </ul>
</li>
<li>
  <a href="data.html">
    <span class="fas fa-download"></span>
     
    Data
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa-book"></span>
     
    CSS Course
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Basic_Molecular.html">Basic Molecular</a>
    </li>
    <li>
      <a href="Editing_Sequence.html">Editing Sequence</a>
    </li>
    <li>
      <a href="DNA_Barcoding.html">DNA Barcoding</a>
    </li>
    <li>
      <a href="Genetika_Populasi.html">Population Genetic</a>
    </li>
    <li>
      <a href="Analisis_Amova.html">Amova Analysis</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa-question fa-lg"></span>
     
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="index.html#faq">FAQ</a>
    </li>
    <li>
      <a href="help.html">Further resources</a>
    </li>
    <li>
      <a href="https://github.com/metabarcodingcourse/2020Course">Source code for this site</a>
    </li>
  </ul>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Visualizing and Annotating Phylogenetic Trees with R+ggtree</h1>

</div>


<!-- This lesson demonstrates how to use **ggtree**, an extension of the ggplot2 package to visualize and annotate phylogenetic trees. Many of the examples here were modified from the [ggtree vignettes](http://bioconductor.org/packages/release/bioc/html/ggtree.html). -->
<!-- _This lesson assumes a [basic familiarity with R](r-basics.html), [data frames](r-dataframes.html), [manipulating data with dplyr and `%>%`](r-dplyr-yeast.html), and most importantly, **[data visualization with ggplot2](r-viz-gapminder.html)**._ -->
<!-- This lesson does _not_ cover methods and software for _generating_ phylogenetic trees, nor does it it cover _interpreting_ phylogenies. **[Here's a quick primer on how to read a phylogeny](http://epidemic.bio.ed.ac.uk/how_to_read_a_phylogeny)** that you should definitely review prior to this lesson, but it is by no means extensive. Genome-wide sequencing allows for examination of the entire genome, and from this, many methods and software tools exist for comparative genomics using SNP- and gene-based phylogenetic analysis, either from unassembled sequencing reads, draft assemblies/contigs, or complete genome sequences. These methods are beyond the scope of this lesson. -->
<!-- ## The `ggtree` Package -->
<!-- **ggtree** is an R package that extends ggplot2 for visualizating and annotating phylogenetic trees with their covariates and other associated data. It is available from [Bioconductor](http://www.bioconductor.org/). Bioconductor is a project to provide tools for analyzing and annotating various kinds of genomic data. You can search and browse Bioconductor packages [here](http://www.bioconductor.org/packages/release/BiocViews.html#___Software). -->
<!-- 1. **ggtree Bioconductor page**: [bioconductor.org/packages/ggtree](http://bioconductor.org/packages/release/bioc/html/ggtree.html). -->
<!-- 1. **ggtree homepage**: [guangchuangyu.github.io/ggtree](https://guangchuangyu.github.io/ggtree/) (contains more information about the package, more documentation, a gallery of beautiful images, and links to related resources). -->
<!-- 1. **ggtree publication**: Yu, Guangchuang, et al. "ggtree: an r package for visualization and annotation of phylogenetic trees with their covariates and other associated data." _Methods in Ecology and Evolution_ (2016) [DOI:10.1111/2041-210X.12628](http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/full). -->
<!-- Bioconductor packages usually have great documentation in the form of _vignettes_. Take a look at the [landing page for ggtree](http://bioconductor.org/packages/release/bioc/html/ggtree.html) -- about halfway down the page under the "Documentation" heading there are multiple walkthrough tutorials directed to different applications and functionalities of ggtree, chock full of runnable examples and explanations. -->
<!-- Just like R packages from CRAN, you only need to install Bioconductor packages once ([instructions here](setup.html#bioconductor)), then load them every time you start a new R session. Load the tidyverse package first. -->
<!-- ```{r load_ggtree_without_junk, echo=FALSE} -->
<!-- suppressPackageStartupMessages(suppressWarnings(library(tidyverse))) -->
<!-- suppressPackageStartupMessages(suppressWarnings(library(ggtree))) -->
<!-- ``` -->
<!-- _A note on masked functions_: Take a second and look through some of the output that you see when you load ggtree after tidyverse. When you first installed ggtree it may have taken a while, because ggtree _depends_ on a number of other R packages. Each of these, in turn, may depend on other packages. These are all loaded into your working environment when you load ggtree. Also notice the lines that start with `The following objects are masked from 'package:...`. One example of this is the `collapse()` function from dplyr. When ggtree was loaded, it loaded it's own function called `collapse()`. Now, if you wanted to use dplyr's collapse function, you'll have to call it explicitly using this kind of syntax: `dplyr::collapse()`. [See this Q&A thread for more](http://stackoverflow.com/questions/4879377/r-masked-functions). -->
<!-- ## Tree Import -->
<!-- From the [ggtree landing page](http://bioconductor.org/packages/release/bioc/html/ggtree.html) take a look at the [Tree Data Import vignette](http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeImport.html). There are many different software packages for creating phylogenetic trees from different types of data, and there are many formats for storing the resulting phylogenetic trees they produce.  -->
<!-- Most tree viewer software (including `R` packages) focus on **Newick** and **Nexus** file formats, and other evolution analysis software might also contain supporting evidence within the file that are ready for annotating a phylogenetic tree. ggtree supports several file formats, including: -->
<!-- - [Newick](https://en.wikipedia.org/wiki/Newick_format) -->
<!-- - [Nexus](https://en.wikipedia.org/wiki/Nexus_file) -->
<!-- - [Phylip](https://en.wikipedia.org/wiki/PHYLIP) -->
<!-- - [Jplace](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0031009) -->
<!-- - [New Hampshire eXtended format (NHX)](https://home.cc.umanitoba.ca/~psgendb/doc/atv/NHX.pdf) -->
<!-- and software output from: -->
<!-- - [BEAST](http://beast2.org/) -->
<!-- - [EPA](http://sco.h-its.org/exelixis/web/software/epa/index.html) -->
<!-- - [HYPHY](http://hyphy.org/w/index.php/Main_Page) -->
<!-- - [PAML](http://abacus.gene.ucl.ac.uk/software/paml.html) -->
<!-- - [PHYLDOG](http://pbil.univ-lyon1.fr/software/phyldog/) -->
<!-- - [pplacer](http://matsen.fhcrc.org/pplacer/) -->
<!-- - [r8s](http://loco.biosci.arizona.edu/r8s/) -->
<!-- - [RAxML](http://sco.h-its.org/exelixis/web/software/raxml/) -->
<!-- - [RevBayes](http://revbayes.github.io/intro.html) -->
<!-- The `ggtree` package implement several parser functions, including: -->
<!-- - `read.tree` for reading Newick files. -->
<!-- - `read.phylip` for reading Phylip files. -->
<!-- - `read.jplace` for reading Jplace files. -->
<!-- - `read.nhx` for reading NHX files. -->
<!-- - `read.beast` for parsing output of [BEAST](http://beast2.org/) -->
<!-- - `read.codeml` for parsing output of [CODEML](http://abacus.gene.ucl.ac.uk/software/paml.html) (`rst` and `mlc` files) -->
<!-- - `read.codeml_mlc` for parsing `mlc` file (output of `CODEML`) -->
<!-- - `read.hyphy` for parsing output of [HYPHY](http://hyphy.org/w/index.php/Main_Page) -->
<!-- - `read.jplace` for parsing `jplace` file including output from [EPA](http://sco.h-its.org/exelixis/web/software/epa/index.html) and [pplacer](http://matsen.fhcrc.org/pplacer/) -->
<!-- - `read.nhx` for parsing `NHX` file including output from [PHYLODOG](http://pbil.univ-lyon1.fr/software/phyldog/) and [RevBayes](http://revbayes.github.io/intro.html) -->
<!-- - `read.paml_rst` for parsing `rst` file (output of `BASEML` and `CODEML`) -->
<!-- - `read.r8s` for parsing output of [r8s](loco.biosci.arizona.edu/r8s/) -->
<!-- - `read.raxml` for parsing output of [RAxML](http://sco.h-its.org/exelixis/web/software/raxml/) -->
<div id="basic-trees" class="section level2">
<h2>Basic trees</h2>
<p>Let’s first import our tree data. We’re going to work with a made-up phylogeny with 13 samples (“tips”). Download the <a href="data/tree_newick.nwk"><strong>tree_newick.nwk</strong> data by clicking here</a> or using the link above. Let’s load the libraries you’ll need if you haven’t already, and then import the tree using <code>read.tree()</code>. Displaying the object itself really isn’t useful. The output just tells you a little bit about the tree itself.</p>
<pre class="r"><code>library(tidyverse)
library(ggtree)
library(phylobase)
library(ape)
tree &lt;- read.tree(&quot;data/tree_newick.nwk&quot;)
tree</code></pre>
<!-- Just like with ggplot2 we created a basic canvas with `ggplot(...)` and added layers with `+geom_???()`, we can do the same here. The ggtree package gives us a `geom_tree()` function. Because ggtree is built on top of ggplot2, you get ggplot2's default gray theme with white lines. You can override this with a theme from the ggtree package. -->
<!-- Because you'll almost always want to add a tree geom and remove the default background and axes, the `ggtree()` function is essentially a shortcut for `ggplot(...) + geom_tree() + theme_tree()`. -->
<pre class="r"><code># build a ggplot with a geom_tree
ggplot(tree) + geom_tree() + theme_tree()</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/first_tree-1.png" width="672" /></p>
<pre class="r"><code># This is convenient shorthand
ggtree(tree)</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/first_tree1-1.png" width="672" /></p>
<p><strong>Adding a Scale Bar</strong></p>
<p>There’s also the treescale geom, which adds a scale bar, or alternatively, you can change the default <code>ggtree()</code> theme to <code>theme_tree2()</code>, which adds a scale on the x-axis.</p>
<ul>
<li>The horizontal dimension in this plot shows the amount of genetic change, and the branches and represent evolutionary lineages changing over time.<br />
</li>
<li>The longer the branch in the horizonal dimension, the larger the amount of change, and the scale tells you this. The units of branch length are usually nucleotide substitutions per site – that is, the number of changes or substitutions divided by the length of the sequence (alternatively, it could represent the percent change, i.e., the number of changes per 100 bases). See <a href="http://epidemic.bio.ed.ac.uk/how_to_read_a_phylogeny">this article</a> for more.</li>
</ul>
<pre class="r"><code># add a scale
ggtree(tree) + geom_treescale()</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/scales-1.png" width="672" /></p>
<pre class="r"><code># or add the entire scale to the x axis with theme_tree2()
ggtree(tree) + theme_tree2()</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/scales1-1.png" width="672" /></p>
<p><strong>Removing Scale Bar and Convert into Phylogram</strong></p>
<p>The default is to plot a phylogram, where the x-axis shows the genetic change / evolutionary distance. If you want to disable scaling and produce a cladogram instead, set the <code>branch.length=&quot;none&quot;</code> option inside the <code>ggtree()</code> call. See <code>?ggtree</code> for more.</p>
<pre class="r"><code>ggtree(tree, branch.length=&quot;none&quot;)</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/cladogram-1.png" width="672" /></p>
<p>The <code>...</code> option in the help for <code>?ggtree</code> represents additional options that are further passed to <code>ggplot()</code>. You can use this to change aesthetics of the plot. Let’s draw a cladogram (no branch scaling) using thick blue dotted lines (note that I’m not mapping these aesthetics to features of the data with <code>aes()</code> – we’ll get to that later).</p>
<pre class="r"><code>ggtree(tree, branch.length=&quot;none&quot;, color=&quot;blue&quot;, size=1, linetype=6)</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/aesthetics-1.png" width="672" /></p>
<hr />
<!-- ### Exercise 1 -->
<p><strong>More Features in Tree Shapes</strong></p>
<p>Look at the help again for <code>?ggtree</code>, specifically at the <code>layout=</code> option. By default, it produces a rectangular layout.</p>
<ol style="list-style-type: decimal">
<li>Create a slanted phylogenetic tree.</li>
<li>Create a circular phylogenetic tree.</li>
<li>Create a circular unscaled cladogram with thick red lines.</li>
</ol>
<pre class="r"><code>suppressWarnings(suppressPackageStartupMessages(library(ggtree)))
tree &lt;- read.tree(file.path(rprojroot::find_rstudio_root_file(), &quot;data&quot;, &quot;tree_newick.nwk&quot;))
ggtree(tree, layout=&quot;slanted&quot;)</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/ex1-1.png" width="672" /></p>
<pre class="r"><code>ggtree(tree, layout=&quot;circular&quot;)</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/ex1-2.png" width="672" /></p>
<pre class="r"><code>ggtree(tree, layout=&quot;circular&quot;, branch.length=&quot;none&quot;, color=&quot;red&quot;, size=3)</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/ex1-3.png" width="672" /></p>
<hr />
<div id="other-tree-geoms" class="section level3">
<h3>Other tree geoms</h3>
<p>Let’s add additional layers. As we did in the <a href="r-viz-gapminder.html">ggplot2 lesson</a>, we can create a plot object, e.g., <code>p</code>, to store the basic layout of a ggplot, and add more layers to it as we desire. Let’s add node and tip points. Let’s finally label the tips.</p>
<p><strong>Create the basic plot</strong></p>
<pre class="r"><code># create the basic plot
p &lt;- ggtree(tree)
p</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/node_tip_points-1.png" width="672" /></p>
<p><strong>Creating Nodepoints from The basic Plot</strong></p>
<pre class="r"><code>p + geom_nodepoint()</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/node_tip_points1-1.png" width="672" /> <span class="math display">\[\\[0.05in]\]</span></p>
<p><strong>Creating Tip Points from the Basic Plot</strong></p>
<pre class="r"><code>p + geom_tippoint()</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/node_tip_points2-1.png" width="672" /></p>
<p><strong>Labeling the Tips from the Basic Plot</strong></p>
<pre class="r"><code>p + geom_tiplab()</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/node_tip_points3-1.png" width="672" /></p>
<!-- ### Exercise 2 -->
<!-- ### Exercise 2 -->
<!-- Similar to how we change the aesthetics for the tree inside the `ggtree()` call, we can also change the aesthetics of the points themselves by passing graphical parameters inside the `geom_nodepoint()` or `geom_tippoint()` calls. Create a phylogeny with the following aesthetic characteristics: -->
<!-- - tips labeled in purple -->
<!-- - purple-colored diamond-shape tip points (hint: Google search "R point characters") -->
<!-- - large semitransparent yellow node points (hint: `alpha=`) -->
<!-- - Add a title with `+ ggtitle(...)` -->
<pre class="r"><code>#tree &lt;- read.tree(file.path(rprojroot::find_rstudio_root_file(), &quot;data&quot;, &quot;tree_newick.nwk&quot;))
p &lt;- ggtree(tree) 
p + 
geom_tiplab(color=&quot;darkorchid&quot;, size=5) + 
geom_tippoint(color=&quot;darkorchid&quot;, size=2, shape=18) + 
geom_nodepoint(color=&quot;goldenrod&quot;, size=4, alpha=1/2) + 
ggtitle(&quot;Not the prettiest phylogenetic aesthetics, but it&#39;ll do.&quot;)</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/ex2-1.png" width="672" /></p>
</div>
</div>
<div id="tree-annotation" class="section level2">
<h2>Tree annotation</h2>
<p>The <code>geom_tiplab()</code> function adds some very rudimentary annotation. Let’s take annotation a bit further. See the <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeAnnotation.html">tree annotation</a> and <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/advanceTreeAnnotation.html">advanced tree annotation</a> vignettes for more.</p>
<div id="internal-node-number" class="section level3">
<h3>Internal node number</h3>
<p>Before we can go further we need to understand how ggtree is handling the tree structure internally. Some of the functions in ggtree for annotating clades need a parameter specifying the internal node number. To get the internal node number, user can use <code>geom_text</code> to display it, where the label is an aesthetic mapping to the “node variable” stored inside the tree object (think of this like the continent variable inside the gapminder object). We also supply the <code>hjust</code> option so that the labels aren’t sitting right on top of the nodes. Read more about this process in the <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeManipulation.html#internal-node-number">ggtree manipulation vignette</a>.</p>
<pre class="r"><code>ggtree(tree) + geom_text(aes(label=node), hjust=-.3)</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/nodenumber-1.png" width="672" /></p>
<p>Another way to get the internal node number is using <code>MRCA()</code> function by providing a vector of taxa names (created using <code>c(&quot;taxon1&quot;, &quot;taxon2&quot;)</code>).. The function will return node number of input taxa’s most recent commond ancestor (MRCA). First, re-create the plot so you can choose which taxa you want to grab the MRCA from.</p>
<pre class="r"><code>ggtree(tree) + geom_tiplab()</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/tiplab_for_mrca-1.png" width="672" /></p>
<p>Let’s grab the most recent common ancestor for taxa C+E, and taxa G+H. We can use <code>MRCA()</code> to get the internal node numbers. Go back to the node-labeled plot from before to confirm this.</p>
<pre class="r"><code>MRCA(tree, tip=c(&quot;C&quot;, &quot;E&quot;))</code></pre>
<pre><code>## [1] 17</code></pre>
<pre class="r"><code>MRCA(tree, tip=c(&quot;G&quot;, &quot;H&quot;))</code></pre>
<pre><code>## [1] 21</code></pre>
<pre class="r"><code>MRCA(tree, tip=c(&quot;L&quot;, &quot;I&quot;))</code></pre>
<pre><code>## [1] 23</code></pre>
</div>
<div id="labeling-clades" class="section level3">
<h3>Labeling clades</h3>
<p>We can use <code>geom_cladelabel()</code> to add another geom layer to annotate a selected clade with a bar indicating the clade with a corresponding label. You select the clades using the internal node number for the node that connects all the taxa in that clade. See the <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeAnnotation.html#annotate-clades">tree annotation vignette</a> for more.</p>
<p>Let’s annotate the clade with the most recent common ancestor between taxa C and E (internal node 17). Let’s make the annotation red. See <code>?geom_cladelabel</code> help for more.</p>
<pre class="r"><code>ggtree(tree) + 
  geom_cladelabel(node=17, label=&quot;Some random clade&quot;, color=&quot;red&quot;)</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/cladelabel-1.png" width="672" /></p>
<p>Let’s add back in the tip labels. Notice how now the clade label is too close to the tip labels. Let’s add an offset to adjust the position. You might have to fiddle with this number to get it looking right.</p>
<pre class="r"><code>ggtree(tree) + 
  geom_tiplab() + 
  geom_cladelabel(node=17, label=&quot;Some random clade&quot;, 
                  color=&quot;red2&quot;, offset=.8)</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/cladelabel_tiplabs2-1.png" width="672" /></p>
<p>Now let’s add another label for the clade connecting taxa G and H (internal node 21).</p>
<pre class="r"><code>ggtree(tree) + 
  geom_tiplab() + 
  geom_cladelabel(node=17, label=&quot;Some random clade&quot;, 
                  color=&quot;red2&quot;, offset=.8) + 
  geom_cladelabel(node=21, label=&quot;A different clade 1&quot;, 
                  color=&quot;blue&quot;, offset=.8)+
  geom_cladelabel(node=23, label=&quot;A different clade 2&quot;, 
                  color=&quot;green&quot;, offset=.8) +
  geom_cladelabel(node=13, label= &quot;Outgroup&quot;, 
                  color=&quot;black&quot;, offset=.8) </code></pre>
<p><img src="r-ggtree_edit_files/figure-html/geom_cladelabel2-1.png" width="672" /></p>
<p>Uh oh. Now we have two problems. First, the labels would look better if they were aligned. That’s simple. Pass <code>align=TRUE</code> to <code>geom_cladelabel()</code> (see <code>?geom_cladelabel</code> help for more). But now, the labels are falling off the edge of the plot. That’s because <code>geom_cladelabel()</code> is just adding it this layer onto the end of the existing canvas that was originally layed out in the ggtree call. This default layout tried to optimize by plotting the entire tree over the entire region of the plot. Here’s how we’ll fix this.</p>
<ol style="list-style-type: decimal">
<li>First create the generic layout of the plot with <code>ggtree(tree)</code>.</li>
<li>Add some tip labels.</li>
<li>Add each clade label.</li>
<li>Remember <strong><code>theme_tree2()</code></strong>? We used it way back to add a scale to the x-axis showing the genetic distance. This is the unit of the x-axis. We need to set the limits on the x-axis. Google around for something like “ggplot2 x axis limits” and you’ll wind up on <a href="https://stackoverflow.com/q/3606697/654296">this StackOverflow page</a> that tells you exactly how to solve it – just add on a <code>+ xlim(..., ...)</code> layer. Here let’s extend out the axis a bit further to the right.</li>
<li>Finally, if we want, we can either <em>comment out</em> the <code>theme_tree2()</code> segment of the code, or we could just add another theme layer on top of the plot altogether, which will override the theme that was set before. <code>theme_tree()</code> doesn’t have the scale.</li>
</ol>
<pre class="r"><code>ggtree(tree) + 
  geom_tiplab() + 
  geom_cladelabel(node=17, label=&quot;Some random clade&quot;, 
                  color=&quot;red2&quot;, offset=.8, align=TRUE) + 
  geom_cladelabel(node=21, label=&quot;A different clade 1&quot;, 
                  color=&quot;blue&quot;, offset=.8, align=TRUE) + 
  geom_cladelabel(node=23, label=&quot;A different clade 2&quot;, 
                  color=&quot;green&quot;, offset=.8) +
  geom_cladelabel(node=13, label= &quot;Outgroup&quot;, 
                  color=&quot;black&quot;, offset=.8) +
  theme_tree2() + 
  xlim(0,70) + 
  theme_tree()</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/geom_cladelabel_final-1.png" width="768" /></p>
<p>Alternatively, we could highlight the entire clade with <code>geom_hilight()</code>. See the help for options to tweak.</p>
<pre class="r"><code>ggtree(tree) + 
  geom_tiplab() + 
  geom_hilight(node=17, fill=&quot;gold&quot;) + 
  geom_hilight(node=21, fill=&quot;purple&quot;)</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/geom_hilight-1.png" width="672" /></p>
<p>Orwe could collapse the entire clade with <code>collapse</code> command. See the help for options to tweak.</p>
<pre class="r"><code>p2 &lt;- p + geom_tiplab()
collapse(p2, 19, &#39;min&#39;, color= &quot;red&quot;, fill=&#39;steelblue&#39;, alpha=.4) %&gt;% 
collapse(24, &#39;max&#39;, fill=&#39;firebrick&#39;, color=&#39;blue&#39;)</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/cladelabel_tiplabs1-1.png" width="672" /></p>
<p><strong>Exercise 1</strong></p>
<ul>
<li>Change the nodes into 18 and 23</li>
<li>Change the outer line of the collapse shape into orange and yellow</li>
</ul>
<p><img src="r-ggtree_edit_files/figure-html/ex6-1.png" width="768" /></p>
</div>
<div id="connecting-taxa" class="section level3">
<h3>Connecting taxa</h3>
<p>Some evolutionary events (e.g. reassortment, horizontal gene transfer) can be visualized with some simple annotations on a tree. The <code>geom_taxalink()</code> layer draws straight or curved lines between any of two nodes in the tree, allow it to show evolutionary events by connecting taxa. Take a look at the <a href="http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/treeAnnotation.html#taxa-connection">tree annotation vignette</a> and <code>?geom_taxalink</code> for more.</p>
<pre class="r"><code>ggtree(tree) + 
  geom_tiplab() + 
  geom_taxalink(&quot;E&quot;, &quot;H&quot;, color=&quot;blue3&quot;) +
  geom_taxalink(&quot;C&quot;, &quot;G&quot;, color=&quot;orange2&quot;, curvature=-.9)</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/unnamed-chunk-1-1.png" width="672" /></p>
<p><strong>Try different values of curvature</strong></p>
<pre class="r"><code>ggtree(tree) + 
  geom_tiplab() + 
  geom_taxalink(&quot;E&quot;, &quot;H&quot;, color=&quot;blue3&quot;) +
  geom_taxalink(&quot;C&quot;, &quot;G&quot;, color=&quot;orange2&quot;, curvature=-.2)</code></pre>
<p><img src="r-ggtree_edit_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<!-- ### Exercise 3 -->
</div>
</div>
<div id="exercise" class="section level2">
<h2>Exercise</h2>
<p>Produce the figure below.</p>
<ol style="list-style-type: decimal">
<li>First, find what the MRCA is for taxa <strong>B+C</strong>, and taxa <strong>L+J</strong>. You can do this in one of two ways:
<ol style="list-style-type: lower-alpha">
<li>Easiest: use <code>MRCA(tree, tip=c(&quot;taxon1&quot;, &quot;taxon2&quot;))</code> for B/C and L/J separately.</li>
<li>Alternatively: use <code>ggtree(tree) + geom_text(aes(label=node), hjust=-.3)</code> to see what the node labels are on the plot. You might also add tip labels here too.</li>
</ol></li>
<li>Draw the tree with <code>ggtree(tree)</code>.</li>
<li>Add tip labels.</li>
<li>Highlight these clades with separate colors.</li>
<li>Add a clade label to the larger superclade (node=17) that we saw before that includes A, B, C, D, and E. You’ll probably need an offset to get this looking right.</li>
<li>Link taxa C to E, and G to J with a dashed gray line (hint: get the geom working first, then try changing the aesthetics. You’ll need <code>linetype=2</code> somewhere in the <code>geom_taxalink()</code>).</li>
<li>Add a scale bar to the bottom by changing the theme.</li>
<li>Add a title.</li>
<li>Optionally, go back to the original <code>ggtree(tree, ...)</code> call and change the layout to <code>&quot;circular&quot;</code>.</li>
</ol>
<p><img src="r-ggtree_edit_files/figure-html/ex3-1.png" width="768" /></p>
<!-- ## Advanced tree annotation -->
<!-- Let's use a previously published dataset from this paper:  -->
<!-- Liang et al. "Expansion of genotypic diversity and establishment of 2009 H1N1 pandemic-origin internal genes in pigs in China." _Journal of virology_ (2014): [88(18):10864-74](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4178866/). -->
<!-- This data was reanalyzed in the [ggtree paper](http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/full). -->
<!-- The subset of the data used here contains 76 H3 hemagglutinin gene sequences of a lineage containing both swine and human influenza A viruses. The sequence data set was re-analyzed by using BEAST (available at <http://beast.bio.ed.ac.uk/>). BEAST (Bayesian Evolutionary Analysis Sampling Trees) can give you rooted, time-measured phylogenies inferred using molecular clock models.  -->
<!-- For this you'll need the `flu_tree_beast.tree` output file from BEAST and the `flu_aasequence.fasta` FASTA file with the multiple sequence alignment. These are both available on the [data downloads page](data.html). First let's read in the tree with `read.beast()` (instead of the `read.tree()` we used before). Let's add a scale bar with theme_tree2(). This gives you genetic distance. But, we have time measured here with molecular clock models. We've only estimated the _relative_ time between branching events, so if we want to actually see _dates_ on the x-axis, we need to supply the most recent sampling date to the `ggtree()` call. Do this by setting `mrsd="YYYY-MM-DD"` inside `ggtree()`.  -->
<!-- Finally, let's add some tip labels. We'll want to right-align them, and by default the dotted line is a little too thick. Let's reduce the `linesize` a bit. Now, some of the labels might be falling off the margin. Set the `xlim` to limit the axis to show between 1990 and 2020. You could get MRCAs and node numbers and do all the annotations that we did before the same way here. -->
<!-- ```{r beastplot, fig.keep="last", fig.width=11, fig.height=11} -->
<!-- # Read the data -->
<!-- #tree <- read.beast("data/flu_tree_beast.tree") -->
<!-- # supply a most recent sampling date so you get the dates -->
<!-- # and add a scale bar -->
<!-- ggtree(tree, mrsd="2013-01-01") +  -->
<!--   theme_tree2()  -->
<!-- # Finally, add tip labels and adjust axis -->
<!-- ggtree(tree, mrsd="2013-01-01") +  -->
<!--   theme_tree2() +  -->
<!--   geom_tiplab(align=TRUE, linesize=.5) +  -->
<!--   xlim(1990, 2020) -->
<!-- ``` -->
<!-- Finally, let's look at `?msaplot`. This puts the multiple sequence alignment and the tree side-by-side. The function takes a tree object (produced with `ggtree()`) and the path to the FASTA multiple sequence alignment. You can do it with the entire MSA, or you could restrict to just a window. Want something interesting-looking, but maybe not all that useful? Try changing the coordinate system of the plot itself by passing `+ coord_polar(theta="y")` to the end of the command! -->
<!-- ```{r msaplot, fig.width=11, fig.height=11} -->
<!-- #msaplot(p=ggtree(tree), fasta="data/flu_aasequence.fasta", window=c(150, 175)) -->
<!-- ``` -->
<!-- Take a look at the [advanced tree annotation vignette](http://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/advanceTreeAnnotation.html) for much, much more! -->
<!-- ## Bonus Exercise -->
<!-- See the [ggtree vignettes](http://bioconductor.org/packages/release/bioc/html/ggtree.html) for more details on how these work. -->
<!-- ### Many trees -->
<!-- ggtree will let you plot many trees at once, and you can facet them the normal ggplot2 way. Let's generate 3 replicates each of 4 random trees with 10, 25, 50, and 100 tips, plotting them all. -->
<!-- ```{r faceted, fig.width=10, fig.height=8} -->
<!-- set.seed(42) -->
<!-- trees <- lapply(rep(c(10, 25, 50, 100), 3), rtree) -->
<!-- class(trees) <- "multiPhylo" -->
<!-- ggtree(trees) + facet_wrap(~.id, scale="free", ncol=4) + ggtitle("Many trees. Such phylogenetics. Wow.") -->
<!-- ``` -->
<!-- ### Plot tree with other data -->
<!-- For showing a phylogenetic tree alongside other panels with your own data, the `facet_plot()` function accepts a input data.frame and a `geom` function to draw the input data.  -->
<!-- ```{r assocdata, fig.width=8} -->
<!-- # Generate a random tree with 30 tips -->
<!-- tree <- rtree(30) -->
<!-- # Make the original plot -->
<!-- p <- ggtree(tree) -->
<!-- # generate some random values for each tip label in the data -->
<!-- d1 <- data.frame(id=tree$tip.label, val=rnorm(30, sd=3)) -->
<!-- # Make a second plot with the original, naming the new plot "dot",  -->
<!-- # using the data you just created, with a point geom. -->
<!-- p2 <- facet_plot(p, panel="dot", data=d1, geom=geom_point, aes(x=val), color='red3') -->
<!-- # Make some more data with another random value. -->
<!-- d2 <- data.frame(id=tree$tip.label, value = abs(rnorm(30, mean=100, sd=50))) -->
<!-- # Now add to that second plot, this time using the new d2 data above,  -->
<!-- # This time showing a bar segment, size 3, colored blue. -->
<!-- p3 <- facet_plot(p2, panel='bar', data=d2, geom=geom_segment,  -->
<!--            aes(x=0, xend=value, y=y, yend=y), size=3, color='blue4')  -->
<!-- # Show all three plots with a scale -->
<!-- p3 + theme_tree2() -->
<!-- ``` -->
<!-- <!-- ### Overlay organism silouhettes -->
<!-- <!-- [phylopic.org](http://phylopic.org/) hosts free silhouette images of animals, plants, and other life forms, all under Creative Commons or Public Domain. You can use ggtree to overlay a phylopic image on your plot at a node of your choosing. Let's show some gram-negative bacteria over the whole plot, and put a _Homo sapiens_ and a dog on those clades we're working with. -->
<!-- ```{r, include=FALSE} -->
<!-- ####################################################################### -->
<!-- knitr::knit_exit() #################################################### -->
<!-- ####################################################################### -->
<!-- ``` -->
</div>

<div class="footer">
<a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank"><i class="fa fa-creative-commons fa-2x"></i></a>&nbsp;
<a href="https://twitter.com/genetics_blog" target="_blank"><i class="fa fa-twitter fa-2x"></i></a>&nbsp;
<a href="https://github.com/4va/biodatasci" target="_blank"><i class="fa fa-github fa-2x"></i></a>&nbsp;
</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
