---
title: "Setup Instructions"
---

## Software

### R

**Install R.** You'll need R version **3.4.0** or higher.[^rversion] Download and install R for [Windows](http://cran.r-project.org/bin/windows/base/) or [Mac](http://cran.r-project.org/bin/macosx/) (download the latest R-3.x.x.pkg file for your appropriate version of Mac OS).

### RStudio

Download and install **[RStudio Desktop](https://www.rstudio.com/products/rstudio/download/)** version >= 1.0.143.

R and RStudio are separate downloads and installations. **R** is the underlying statistical computing environment, but using R alone is no fun. **RStudio** is a graphical integrated development environment that makes using R much easier. You need R installed before you install RStudio.

### CRAN packages

Launch RStudio (RStudio, *not R itself*). Ensure that you have internet access, then copy and paste the following command into the **Console** panel (usually the lower-left panel, by default) and hit the Enter/Return key. 

```r
install.packages("tidyverse")
```


_A few notes_:

- Commands are case-sensitive. 
- You must be connected to the internet.
- The [tidyverse](http://tidyverse.org/) package is kind of a meta-package that automatically installs/loads many core packages that we use throughout the workshops.[^tidyverse]
- Even if you've installed these packages in the past, do re-install the most recent version. Many of these packages are updated often, and we may use new features in the workshop that aren't available in older versions.
- If you're using Windows you might see errors about not having permission to modify the existing libraries -- disregard these. You can avoid this by running RStudio as an administrator (right click the RStudio icon, then click "Run as Administrator").

[^rversion]: R version 3.4.0 was released April 2017. If you have not updated your R installation since then, you need to upgrade to a more recent version, since several of the required packages depend on a version at least this recent. You can check your R version with the `sessionInfo()` command.

[^tidyverse]: Installing/loading the tidyverse **tidyverse** will install/load the core tidyverse packages that you are likely to use in almost every analysis:
**ggplot2** (for data visualisation), **dplyr** (for data manipulation), **tidyr** (for data tidying), **readr** (for data import), **purrr** (for functional programming), and **tibble** (for tibbles, a modern re-imagining of data frames). It also installs a selection of other tidyverse packages that you're likely to use frequently, but probably not in every analysis (these are installed, but you'll have to load them separately with `library(packageName)`). This includes: **hms** (for times), **stringr** (for strings), **lubridate** (for date/times), **forcats** (for factors), **DBI** (for databases), **haven** (for SPSS, SAS and Stata files), **httr** (for web apis), **jsonlite** (or JSON), **readxl** (for .xls and .xlsx files), **rvest** (for web scraping), **xml2** (for XML), **modelr** (for modelling within a pipeline), and **broom** (for turning models into tidy data). After installing tidyverse with `install.packages("tidyverse")` and loading it with `library(tidyverse)`, you can use `tidyverse_update()` to update all the tidyverse packages installed on your system at once.

Check that you've installed everything correctly by closing and reopening RStudio and entering the following commands at the console window (don't worry about the _Conflicts with tidy packages_ warning):

```r
library(tidyverse)
```

This may produce some notes or other output, but as long as you don't get an error message, you're good to go. If you get a message that says something like: `Error in library(somePackageName) : there is no package called 'somePackageName'`, then the required packages did not install correctly. Please do not hesitate to email me _prior to the course_ if you are still having difficulty. In this email, please copy and paste what you typed in the console, and all of the output that streams by in the console.

### Bioconductor

For some classes (e.g., RNA-seq), you'll need to install a few [Bioconductor](http://bioconductor.org/) packages. These packages are installed differently than "regular" R packages from CRAN. Copy and paste these lines of code into your R console **one at a time**. 

```r
source("http://bioconductor.org/biocLite.R")
biocLite()
biocLite("DESeq2")
biocLite("ggtree")
```

A few notes:

- We will be using the latest versions of Bioconductor from the 3.5 release. This requires R version 3.4.0 or higher. If you have R 3.4.0 installed, running the commands above will install Bioconductor 3.5. See <http://bioconductor.org/news/bioc_3_5_release/>.
- If at any point in the Bioconductor package installations you get prompts in the console asking you to update any existing packages, type `n` at the prompt at hit enter.
- If you see a note long the lines of "_binary version available but the source version is later_", followed by a question, "_Do you want to install from sources the package which needs compilation? y/n_", type **`n` for no**, and hit enter.

Check that you've installed everything correctly by closing and reopening RStudio and entering the following commands at the console window, one at a time:

```r
library(DESeq2)
library(ggtree)
```

If you get a message that says something like: `Error in library(somePackageName) : there is no package called 'somePackageName'`, then the required packages did not install correctly. Please do not hesitate to email me _prior to the course_ if you are still having difficulty. In this email, please copy and paste what you typed in the console, and all of the output that streams by in the console.


### RMarkdown

Several additional setup steps required for the reproducible research with RMarkdown class.

1. First, install R, RStudio, and the tidyverse package as described above. Also install the knitr and rmarkdown packages.
```r
install.packages("knitr")
install.packages("rmarkdown")
```
1. Next, launch RStudio (not R). Click File, New File, R Markdown. This may tell you that you need to install additional packages (`knitr`, `yaml`, `htmltools`, `caTools`, `bitops`, `rmarkdown`, and maybe a few others). Click "Yes" to install these.
1. **_Optional_:** If you want to convert to PDF, you will need to install a **$\LaTeX$** typesetting engine. This differs on Mac and Windows. _Note that this part of the installation may take up to several hours, and isn't strictly required for the class._
    - **Windows**: Download and install MiKTeX: <https://miktex.org/download>. Read the installation tutorial first at <https://miktex.org/howto/install-miktex>.
    - **Mac**: Download and install MacTeX.pkg at <http://www.tug.org/mactex/mactex-download.html>. This is a very large download (>2 gigabytes), and may take a while depending on your network speed. Do this _prior to the course_.


## Get Data

Click the **[<i class="fa fa-download"></i> Data](data.html)** link on the navbar at the top. You can download all the data needed by downloading **[this zip file <i class="fa fa-file-archive-o" aria-hidden="true"></i>](data.zip)** or by downloading individual data sets as needed at the **[<i class="fa fa-download"></i> Data](data.html)** page.


## Required Reading

### Tidy Data

1. "For Big-Data Scientists, Janitor Work Is Key Hurdle to Insights." _The New York Times_, August 8, 2014. Available at: <https://nyti.ms/2jNUfHo>.
1. Sections 1-3 of [Wickham, H. "Tidy Data." _Journal of Statistical Software_ 59:10 (2014)](http://www.jstatsoft.org/article/view/v059i10/v59i10.pdf).
1. Click through each section of [stephenturner.us/dataorganization/](http://stephenturner.us/dataorganization/).

### RMarkdown Background

Spend a few minutes to learn a little bit about _Markdown_. All you really need to know is that Markdown is a lightweight markup language that lets you create styled text (like **bold**, _italics_, [links](http://bioconnector.org/markdown), etc.) using a very lightweight plain-text syntax: `(like **bold**, _italics_, [links](http://bioconnector.org/markdown), etc.)`. The resulting text file can be _rendered_ into many downstream formats, like PDF (for printing) or HTML (websites).

1. _(30 seconds)_ Read the summary paragraph on the [Wikipedia page](https://en.wikipedia.org/wiki/Markdown). 
1. _(1 minute)_ Bookmark and refer to this markdown reference: <http://commonmark.org/help/>.
1. _(5-10 minutes)_ Run through this 10-minute in-browser markdown tutorial: <http://commonmark.org/help/tutorial/>.
1. _(5-10 minutes)_ Go to <http://dillinger.io/>, an in-browser Markdown editor, and play around. Write a simple markdown document, and export it to HTML and/or PDF.
1. _(10 minutes)_ See RStudio's excellent documentation on Rmarkdown at <http://rmarkdown.rstudio.com/>. Click "Getting Started" and watch the 1 minute video on the [Introduction page](http://rmarkdown.rstudio.com/lesson-1.html). Continue reading through each section here on the navigation bar to the left (_Introduction_ through _Cheatsheets_, and optionally download and print out the cheat sheet). Finally, browse through the [RMarkdown Gallery](http://rmarkdown.rstudio.com/gallery.html). 
1. _(0 seconds)_ No need to look now, but don't forget that the course [help page](help.html) has some [useful resources on Markdown+RMarkdown](help.html#markdown-rmarkdown-resources).


### Phylogenetic trees

1. How to read a phylogeny [epidemic.bio.ed.ac.uk/how_to_read_a_phylogeny](http://epidemic.bio.ed.ac.uk/how_to_read_a_phylogeny).

### RNA-seq

1. [Conesa et al. A survey of best practices for RNA-seq data analysis. _Genome Biology_ 17:13 (2016)](http://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0881-8).
1. [Soneson et al. "Differential analyses for RNA-seq: transcript-level estimates improve gene-level inferences." _F1000Research_ 4 (2015)](https://f1000research.com/articles/4-1521/v2).
1. Abstract and introduction sections of [Himes et al. "RNA-Seq transcriptome profiling identifies CRISPLD2 as a glucocorticoid responsive gene that modulates cytokine function in airway smooth muscle cells." _PLoS ONE_ 9.6 (2014): e99625](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0099625).


----
